packages/langchain/src/orchestration/client.test.ts

```diff
@@ -129,4 +158,93 @@ describe('orchestration service client', () => {
       'Request failed with status code 400'
     );
   }, 1000);
+
+  it('supports streaming responses', async () => {
+    mockStreamInferenceWithResilience();
+
+    const client = new OrchestrationClient(config);
+    const stream = await client.stream([]);
+    let iterations = 0;
+    const maxIterations = 2;
+    let intermediateChunk: AIMessageChunk | undefined;
+
+    for await (const chunk of stream) {
+      iterations++;
+      intermediateChunk = !intermediateChunk
+        ? chunk
+        : intermediateChunk.concat(chunk);
+      if (iterations >= maxIterations) {
+        break;
+      }
+    }
+    expect(intermediateChunk).toBeDefined();
+    expect(intermediateChunk!.content).toBeDefined();
+    expect(intermediateChunk!.content).toEqual(
+      'The SAP Cloud SDK is a comprehensive development toolkit designed to simplify and accelerate the cre'
+    );
+  });
+
+  it('streaming with abort signal', async () => {
+    mockStreamInferenceWithResilience();
+    const client = new OrchestrationClient(config);
+    const controller = new AbortController();
+    const { signal } = controller;
+    const stream = await client.stream([], { signal });
+    const streamPromise = async () => {
+      for await (const _chunk of stream) {
+        controller.abort();
+      }
+    };
+
+    await expect(streamPromise()).rejects.toThrow();
+  }, 1000);
+
+  it('streaming with callbacks', async () => {
+    mockStreamInferenceWithResilience();
+    let tokenCount = 0;
+    const callbackHandler = {
+      handleLLMNewToken: jest.fn().mockImplementation(() => {
+        tokenCount += 1;
+      })
+    };
+    const client = new OrchestrationClient(config, {
+      callbacks: [callbackHandler]
+    });
+    const stream = await client.stream([]);
+    const chunks = [];
+
+    for await (const chunk of stream) {
+      chunks.push(chunk);
+      break;
+    }
+    expect(callbackHandler.handleLLMNewToken).toHaveBeenCalled();
+    const firstCallArgs = callbackHandler.handleLLMNewToken.mock.calls[0];
+    expect(firstCallArgs).toBeDefined();
+    // First chunk content is empty
+    expect(firstCallArgs[0]).toEqual('');
+    // Second argument should be the token indices
+    expect(firstCallArgs[1]).toEqual({ prompt: 0, completion: 0 });
```

Comment:
[q] I do not understand this honestly. The mock handler function here doesnt seem to take any arguments. Where do they come from?
[q] Why check : expect(chunks.length).toBeGreaterThan(0); Won't it just be one? What is the point of checking this value even wrt callbacks?

Reply:
First question: Even though the call back function, doesn't take arguments here, when the following snippet is executed in the `_streamResponseChunks`  function, these arguments gets passed to the callback for each chunk.
```
await runManager?.handleLLMNewToken(
 content,
 tokenIndices,
 undefined,
 undefined,
 undefined,
 { chunk: generationChunk }
);
 ```      
Second question: You are right, we don't need to check this value here.